lxd/instance.go

```diff
@@ -577,6 +593,13 @@ func pruneExpiredContainerSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
 		}
 
 		opRun := func(op *operations.Operation) error {
+			defer func() {
+				snapshotsMutex.Lock()
+				for _, snapshot := range expiredSnapshots {
+					delete(snapshotsInProgress, snapshot.ID())
+				}
+				snapshotsMutex.Unlock()
+			}()
 			return pruneExpiredContainerSnapshots(ctx, d, expiredSnapshots)
```

Comment:
I think it would be cleaner to add/remove/check entries in snapshotsInProgress from inside `pruneExpiredContainerSnapshots` rather than `pruneExpiredContainerSnapshotsTask`, as that is where the actual `snapshot.Delete()` function that we want to skip is called.

Reply:
This part of code was removed. Look at the comment below.